entgql/schema.go

```diff
@@ -319,6 +328,59 @@ func (e *schemaGenerator) buildTypeFields(t *gen.Type) (ast.FieldList, error) {
 	return fields, nil
 }
 
+func (e *schemaGenerator) buildEdge(edge *gen.Edge, edgeAnt *Annotation) ([]*ast.FieldDefinition, error) {
+	gqlType, ant, err := gqlTypeFromNode(edge.Type)
+	if err != nil {
+		return nil, err
+	}
+
+	edgeField := camel(edge.Name)
+	var mappings []string
+	if len(edgeAnt.Mapping) > 0 {
+		mappings = edgeAnt.Mapping
+	} else {
+		mappings = []string{edgeField}
+	}
+
+	var fields []*ast.FieldDefinition
+	for _, name := range mappings {
+		fieldDef := &ast.FieldDefinition{
+			Name:       name,
+			Directives: e.buildDirectives(edgeAnt.Directives),
+		}
+		if name != edgeField {
+			fieldDef.Directives = append(fieldDef.Directives, goField(edgeField))
+		}
+		switch {
+		case edge.Unique:
+			fieldDef.Type = namedType(gqlType, edge.Optional)
+		case edgeAnt.RelayConnection:
+			if !e.relaySpec {
+				return nil, ErrRelaySpecDisabled
+			}
+			if !ant.RelayConnection {
+				return nil, fmt.Errorf("entgql: must enable Relay Connection via the entgql.RelayConnection annotation on the %s entity", edge.Type.Name)
+			}
```

Comment:
qq about this - in case a Node is marked as `RelayConnection` in its schema, do we add to the Query object?